x-pack/plugins/observability/public/embeddable/slo/alerts/components/slo_alerts_table.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { AlertConsumers } from '@kbn/rule-data-utils';
+import type { TimeRange } from '@kbn/es-query';
+import { SloEmbeddableDeps } from '../slo_alerts_embeddable';
+import type { SloItem } from '../types';
+
+type SloIdAndInstanceId = [string, string];
+const ALERTS_PER_PAGE = 10;
+const ALERTS_TABLE_ID = 'xpack.observability.sloAlertsEmbeddable.alert.table';
+
+interface Props {
+  deps: SloEmbeddableDeps;
+  slos: SloItem[];
+  timeRange: TimeRange;
+  onLoaded?: () => void;
+}
+
+export function SloAlertsTable({ slos, deps, timeRange, onLoaded }: Props) {
+  const {
+    triggersActionsUi: { alertsTableConfigurationRegistry, getAlertsStateTable: AlertsStateTable },
+  } = deps;
+  const slosWithoutName = slos.map((slo) => ({
+    id: slo.id,
+    instanceId: slo.instanceId,
+  }));
+  const sloIdsAndInstanceIds = slosWithoutName.map(Object.values) as SloIdAndInstanceId[];
+
+  return (
+    <AlertsStateTable
+      query={{
```

Comment:
I will wrap the query in a function with `useMemo`

Reply:
i think we can simplify just like that
```ts
function getSloAlertTableQuery(slos: SloItem[], timeRange: TimeRange) {
  return {
    bool: {
      filter: [
        {
          range: {
            '@timestamp': {
              gte: timeRange.from,
            },
          },
        },
        {
          term: {
            'kibana.alert.rule.rule_type_id': 'slo.rules.burnRate',
          },
        },
      ],
      should: slos.map((slo) => ({
        bool: {
          filter: [{ term: { 'slo.id': slo.id } }, { term: { 'slo.instanceId': slo.instanceId } }],
        },
      })),
      minimum_should_match: 1,
    },
  };
}

export function SloAlertsTable({ slos, deps, timeRange, onLoaded }: Props) {
  const {
    triggersActionsUi: { alertsTableConfigurationRegistry, getAlertsStateTable: AlertsStateTable },
  } = deps;
  const alertQuery = useMemo(() => getSloAlertTableQuery(slos, timeRange), [slos, timeRange]);

  return (
    <AlertsStateTable
      query={alertQuery}
      alertsTableConfigurationRegistry={alertsTableConfigurationRegistry}
      configurationId={AlertConsumers.SLO}
      featureIds={[AlertConsumers.SLO, AlertConsumers.OBSERVABILITY]}
      hideLazyLoader
      id={ALERTS_TABLE_ID}
      pageSize={ALERTS_PER_PAGE}
      showAlertStatusWithFlapping
      onLoaded={() => {
        if (onLoaded) {
          onLoaded();
        }
      }}
    />
  );
}
```

Reply:
then you can share this utils function `getSloAlertQuery` between `slo_alerts_table.tsx` and `slo_alerts_summary`

Reply:
@XavierM Nice suggestion and it makes sense. Unfortunately above query doesn't work well with alerts embeddable table. It brings more alerts than it should. I have created [another ticket](https://github.com/elastic/kibana/issues/172448) to tackle this separately, cause it needs further investigation plus automated testing. 

Just to clarify something. Above query is used in the alert summary calculation and there it works well. It doesn't work well with the alerts table. As part of the ticket I created, I want to implement your suggestion, so that both alert summary and alerts table use the same ES query

Reply:
Also, I wanted to add that with server-less. Everything we merged in main should be working as the user will expect it.

Reply:
Regarding serverless I pass `isServerless`, plus I added `kibanaVersion` in this commit as [well](https://github.com/elastic/kibana/pull/169910/commits/62d4242d0c1dff3e5bf534339131900879a861c7#diff-c0d25395e857d8c4c01b74474990de8934f773268a34aaf89cca8874a0ff8e75R112). Anything else you think I should take into consideration regarding serverless?